pkg/common/cns-lib/volume/util.go

```diff
@@ -161,3 +166,154 @@ func updateQueryResult(ctx context.Context, m *defaultManager, res *cnstypes.Cns
+	var task *object.Task
+	log := logger.GetLogger(ctx)
+	taskDetailsInMap, ok := volumeTaskMap[volNameFromInputSpec]
+	if ok {
+		task = taskDetailsInMap.task
+		log.Infof("CreateVolume task still pending for Volume: %q, with taskInfo: %+v",
+			volNameFromInputSpec, task)
+	}
+	return task
+}
+
+// invokeCNSCreateVolume truncates the input volume name and invokes a CreateVolume operation for that volume on CNS.
+func invokeCNSCreateVolume(ctx context.Context, virtualCenter *cnsvsphere.VirtualCenter,
+	spec *cnstypes.CnsVolumeCreateSpec) (*object.Task, error) {
+	var cnsCreateSpecList []cnstypes.CnsVolumeCreateSpec
+	log := logger.GetLogger(ctx)
+	// Truncate the volume name to make sure the name is within 80 characters before calling CNS.
+	if len(spec.Name) > maxLengthOfVolumeNameInCNS {
+		volNameAfterTruncate := spec.Name[0 : maxLengthOfVolumeNameInCNS-1]
+		log.Infof("Create Volume with name %s is too long, truncate it to %s", spec.Name, volNameAfterTruncate)
+		spec.Name = volNameAfterTruncate
+		log.Debugf("CNS Create Volume is called with %v", spew.Sdump(*spec))
+	}
+	cnsCreateSpecList = append(cnsCreateSpecList, *spec)
+	task, err := virtualCenter.CnsClient.CreateVolume(ctx, cnsCreateSpecList)
+	if err != nil {
+		log.Errorf("CNS CreateVolume failed from vCenter %q with err: %v", virtualCenter.Config.Host, err)
+		return nil, err
+	}
+	return task, nil
+}
+
+// isStaticallyProvisioned returns true if the input spec is for a statically provisioned volume.
+func isStaticallyProvisioned(spec *cnstypes.CnsVolumeCreateSpec) bool {
+	var isStaticallyProvisionedBlockVolume bool
+	var isStaticallyProvisionedFileVolume bool
+	if spec.VolumeType == string(cnstypes.CnsVolumeTypeBlock) {
+		blockBackingDetails, ok := spec.BackingObjectDetails.(*cnstypes.CnsBlockBackingDetails)
+		if ok && (blockBackingDetails.BackingDiskId != "" || blockBackingDetails.BackingDiskUrlPath != "") {
+			isStaticallyProvisionedBlockVolume = true
+		}
+	}
+	if spec.VolumeType == string(cnstypes.CnsVolumeTypeFile) {
+		fileBackingDetails, ok := spec.BackingObjectDetails.(*cnstypes.CnsVsanFileShareBackingDetails)
+		if ok && fileBackingDetails.BackingFileId != "" {
+			isStaticallyProvisionedFileVolume = true
+		}
+	}
+	return isStaticallyProvisionedBlockVolume || isStaticallyProvisionedFileVolume
+}
+
+// getTaskResultFromTaskInfo returns the task result for a given task.
+func getTaskResultFromTaskInfo(ctx context.Context, taskInfo *types.TaskInfo) (cnstypes.BaseCnsVolumeOperationResult,
+	error) {
+	log := logger.GetLogger(ctx)
+
+	// Get the taskResult.
+	taskResult, err := cns.GetTaskResult(ctx, taskInfo)
+
+	if err != nil {
+		log.Errorf("failed to get CreateVolume result for task with ID: %q, opId: %q result: %+v",
+			taskInfo.Task.Value, taskInfo.ActivationId, taskResult)
+		return nil, err
+	}
+
+	if taskResult == nil {
+		log.Errorf("taskResult is empty for CreateVolume task: %q", taskInfo.ActivationId)
```

Comment:
Remove `CreateVolume` if we are planning to use the function for other volume operations

Reply:
done